src/flamenco/vm/fd_vm_interp_core.c

```diff
@@ -870,3 +872,104 @@ sigexit:     FD_VM_INTERP_FAULT; /* cu current */ /* err current */         goto
 # if defined(__GNUC__)
 # pragma GCC diagnostic pop
 # endif
+
+/*   Agave/JIT CU model analysis (and why we are conformant!):
+
+     The Agave JIT employs a similar strategy of accumulating instructions
+     in a linear run and processing them at the start of a new linear 
+     run/branch (side note: the JIT treats the LDQ instruction as a "branch" 
+     that jumps pc + 2). 
+     
+     In what is assumed to be an act of register conservation, the JIT 
+     uses a catch-all "instruction meter" (IM) register (REGISTER_INSTRUCTION_METER)
+     that represents two different interpretations of the question
+     "how many instructions can I execute?" depending on the context.
+
+     The IM either represents:
+        1. IM => The number of instructions remaining before exhausting CU 
+        budget.
+        2. IM' => The last pc you can execute in the current linear run before
+        exhausting CU budget.  Mathematically, IM' = IM + pc0
+        where pc0, just like our definition, is the start of the linear run 
+        
+        Note: IM' can go past the actual basic block/segment. In-fact, 
+        it typically does, and implies we can execute the full block without
+        exhausting CU budget (reminder that LDQ is treated as a branch).
+      
+      By default, the IM' form is used during execution. The IM form is used:
+        - (transiently) during the processing of a branch instruction 
+        - in post-VM cleanup (updates EbpfVm::previous_instruction_meter).
+
+      When a branch instruction is encountered, the JIT checks
+      for CU exhaustion with pc > IM', and throws an exception if so. Otherwise,
```

Comment:
```suggestion
      for CU exhaustion with pc > IM', and throws an exception if so. This is valid, because as described above,
      IM' is the largest PC you can reach.
      
      If we haven't exhausted our CU limit,
```